### PR TITLE
check config error in status endpoint

### PR DESCRIPTION
**Pull request by nurali-techie**
Fixes: https://github.com/fabric8-services/fabric8-env/issues/12

### CODE REVIEW

File: configuration/configuration.go [hunk 1]
```diff
@@ -74,19 +64,41 @@ func New(configFilePath string) (*Registry, error) {
...
+	allSettings := c.v.AllSettings()
+	y, err := yaml.Marshal(&allSettings)
+	if err != nil {
+		log.WithFields(map[string]interface{}{
+			"settings": allSettings,
+			"err":      err,
+		}).Panicln("Failed to marshall config to string")
 	}
-	return envConfigPath
+	return fmt.Sprintf("%s\n", y)
```
Comment (xcoulon) [inline configuration/configuration.go:101]: it could be just `return string(y)` here

**Review by xcoulon — CHANGES_REQUESTED** [review]
looks good, but I think that you did not test all cases on config error. See https://github.com/fabric8-services/fabric8-cluster/blob/master/configuration/configuration_blackbox_test.go for something more exhaustive.

File: configuration/configuration.go (hunk 1 above)
Comment (nurali-techie) [reply configuration/configuration.go:101]: see there is `\n` due to which `Sprintf` is used.

**Comment by nurali-techie** [comment]
@xcoulon thanks for the look, yup config_test is missed out.  Let me do that :+1:

**Comment by nurali-techie** [comment]
@xcoulon I just checked in details and I see tests are provided for the changes.  This is the effective changes I have done in configuration.go file [changes_link](https://github.com/fabric8-services/fabric8-env/blob/6bae8bba766b7ec5bca13ec5beb83247c63e4a74/configuration/configuration.go#L71-L87) which mainly deal with setting/modifying configErr depending on different config_param value.  Against that I have provided these tests [test_file](https://github.com/fabric8-services/fabric8-env/blob/6bae8bba766b7ec5bca13ec5beb83247c63e4a74/controller/status_blackbox_test.go#L77-L105).  Test check, configErr when devMode=true and devMode=false.

I am not sure what other test I can add .. plz give me test_definition to help me understand.

File: configuration/configuration.go (hunk 1 above)
Comment (xcoulon) [reply configuration/configuration.go:101]: why do you need to add a `\n` here?

**Comment by xcoulon** [comment]
> @xcoulon I just checked in details and I see tests are provided for the changes. This is the effective changes I have done in configuration.go file [changes_link](https://github.com/fabric8-services/fabric8-env/blob/6bae8bba766b7ec5bca13ec5beb83247c63e4a74/configuration/configuration.go#L71-L87) which mainly deal with setting/modifying configErr depending on different config_param value. Against that I have provided these tests [test_file](https://github.com/fabric8-services/fabric8-env/blob/6bae8bba766b7ec5bca13ec5beb83247c63e4a74/controller/status_blackbox_test.go#L77-L105). Test check, configErr when devMode=true and devMode=false.
> 
> I am not sure what other test I can add .. plz give me test_definition to help me understand.

@nurali-techie I think that it would be good if you could provide a set of tests to verify that the configuration reports an error in all cases that are covered by the code, that's all. The most straightforward way to do it would be in a `configuration_test.go`, not at the `status controller` level.

File: configuration/configuration.go (hunk 1 above)
Comment (nurali-techie) [reply configuration/configuration.go:101]: @xcoulon plz note, this method is not part of this PR change, I just moved it in this PR.
Previously, it was copied from f8-wit.  I also checked in f8-cluster, f8-auth, f8-tenant, all has this `\n`.

**Comment by nurali-techie** [comment]
@xcoulon incorporate your review comment, kindly have a look.

**Review by xcoulon — APPROVED** [review]

**Comment by xcoulon** [comment]
> @xcoulon incorporate your review comment, kindly have a look.

thanks for taking the time to add those tests, @nurali-techie! I just approved your PR

**Pull request merged**